modules/lang-painless/src/main/java/org/elasticsearch/painless/antlr/PainlessParser.java

```diff
@@ -2834,8 +2837,21 @@ public final LambdaContext lambda() throws RecognitionException {
       }
       setState(448);
       match(ARROW);
-      setState(449);
-      block();
+      setState(451);
+      switch ( getInterpreter().adaptivePredict(_input,38,_ctx) ) {
```

Comment:
ok, the required semicolon is actually independent of this PR (existing issue). I hit that problem first, and tried to figure it out but couldn't.  So i worked on this issue to at least make progress. Lemme throw some more tests at it to look for real ambiguities.